applications/asset_tracker_v2/src/cloud/nrf_cloud_integration.c

```diff
@@ -175,6 +175,9 @@ static void nrf_cloud_event_handler(const struct nrf_cloud_evt *evt)
 	case NRF_CLOUD_EVT_TRANSPORT_CONNECTED:
 		LOG_DBG("NRF_CLOUD_EVT_TRANSPORT_CONNECTED");
 		break;
+	case NRF_CLOUD_EVT_TRANSPORT_CONNECT_ERROR:
+		LOG_ERR("NRF_CLOUD_EVT_TRANSPORT_CONNECT_ERROR: %d", evt->status);
```

Comment:
I'm not sure but does this work properly in all aspects (like UI module LED states etc) if this doesn't trigger an error? So should we add these here and if not, why?
```
		cloud_wrap_evt.type = CLOUD_WRAP_EVT_ERROR;
		notify = true;
```
As said, I'm not 100% sure this is needed but should be checked carefully because otherwise last sent event (connecting) will remain for a longer period.
@simensrostad could also have input here.

Reply:
It depends.
`CLOUD_WRAP_EVT_ERROR` is intended for irrecoverable errors that the app needs to reboot on in order to recover.
Looking at the various enums in `nrf_cloud_error_status` it seems that simply doing a reboot would be an acceptable _catch all_, since most of them are quite severe.

Reply:
I did also think about this, but kept the behavior as it was earlier so far. Some of the transport connect related errors are such, that retry or reboot will not help. For example, the errors related to certificates. Then again, some are such, that trying again may very well help, for example no response to DNS query.

I don't know if we want to reboot the device in either case?

Reply:
In general, this is how we deal with most errors in ATv2. Its not ideal, but we really don't have any good way of handling different severities. If the behavior is kept I think we can leave it as is.
We could lower the max amount of connection attempts instead, which IMO is set way to high atm. CONFIG_CLOUD_CONNECT_RETRIES.

Reply:
The behavior stays the same. Earlier, the nRF Cloud library sent a second `NRF_CLOUD_EVT_TRANSPORT_CONNECTED` event in case of an error, but ATv2 didn't react to the event, just logged it (not the status, though). Now, the error is indicated with an `NRF_CLOUD_EVT_TRANSPORT_CONNECT_ERROR` event. The event and the status is logged, so the error is more visible to the user. But it still does not trigger immediate retry or reset.

Reply:
I still need to confirm that you are talking about the same thing.

@simensrostad:
> It depends. `CLOUD_WRAP_EVT_ERROR` is intended for irrecoverable errors that the app needs to reboot on in order to recover.

@tokangas:
> I did also think about this, but kept the behavior as it was earlier so far.

@simensrostad:
> In general, this is how we deal with most errors in ATv2. Its not ideal, but we really don't have any good way of handling different severities. If the behavior is kept I think we can leave it as is.

I believe ATv2 developers might have thought that a connection error from nRF Cloud is indicated with `NRF_CLOUD_EVT_ERROR` when this function sets `CLOUD_WRAP_EVT_ERROR`. However, connection error has been indicated with `NRF_CLOUD_EVT_TRANSPORT_CONNECTING` event with the status field set to error value. This hasn't been handled in ATv2 side meaning that ATv2 is in "connecting" state until something else is done. I guess there might be a retry or some ATv2 connection timeout. This is the behavior that this PR will keep as is. And my question is that should connection error trigger `CLOUD_WRAP_EVT_ERROR` rather than do nothing?

Reply:
It looks like ATv2 starts connecting to the cloud and schedules `connect_check_work` to be executed after some time. If cloud connection has not succeeded when the work gets executed, ATv2 tries again. As @simensrostad mentioned, CONFIG_CLOUD_CONNECT_RETRIES is set pretty high, it's 10 retries by default. Also, the time between tries doubles each time (starting from 32 seconds).

So, it's not necessary to do anything when `NRF_CLOUD_EVT_TRANSPORT_CONNECT_ERROR` event is received. The event is just logged to make the error visible in the logs.

Reply:
@simensrostad Was it expected that connection error would do a `CLOUD_WRAP_EVT_ERROR` before or not?

Irrespective of that, do we want `CLOUD_WRAP_EVT_ERROR` to occur in case of connection error? I'm not trying to direct you to any direction. I want to understand what was the intended behavior.

Reply:
I'm having very little time now to browse through the code and cannot do any trials. I guess CLOUD_WRAP_EVT_ERROR for connection error is not good from connection logic point of view. It's better to do re-trials.

I'm not sure if UI module works nicely with this though, or has been working in the past. CONNECTING LEDs will be blinking throughout the retry process which takes a long time. I'm not sure if this is intended or not.

Reply:
CI is green, I'll ask for merge. I can create a new PR to change the SLM logging as you requested.

Reply:
> I'm having very little time now to browse through the code and cannot do any trials. I guess CLOUD_WRAP_EVT_ERROR for connection error is not good from connection logic point of view. It's better to do re-trials.
> 
> I'm not sure if UI module works nicely with this though, or has been working in the past. CONNECTING LEDs will be blinking throughout the retry process which takes a long time. I'm not sure if this is intended or not.

You are right. As long as the device is connecting, only the corresponding LED pattern will be shown.
`STATE_CLOUD_CONNECTING` is a main state in the UI module state machine. But, some events will trigger different patterns. Irrecoverable errors for instance.

I think the idea here was that its easier to see when there is a cloud related connection problem if the connection LED pattern is persisted. It should be easy to add some sub-states that could show roughly what the issue is, atleast the severity of it. But, I think this is something that in general customers should implement themselves depending on their needs.

However, I'm definitely not opposed some improvement here if someone wants to do the work :smile:

Reply:
> I think the idea here was that its easier to see when there is a cloud related connection problem if the connection LED pattern is persisted.

I guess this is fine because there is both retry mechanism and a reboot after retries are done.